tools/apps/src/lib.rs

```diff
@@ -0,0 +1,133 @@
+// Copyright (C) 2020, Cloudflare, Inc.
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are
+// met:
+//
+//     * Redistributions of source code must retain the above copyright notice,
+//       this list of conditions and the following disclaimer.
+//
+//     * Redistributions in binary form must reproduce the above copyright
+//       notice, this list of conditions and the following disclaimer in the
+//       documentation and/or other materials provided with the distribution.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+// IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+// THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+// PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+// LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+// NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+//! Quiche application utilities.
+//!
+//! This module provides some utility functions that are common to quiche
+//! applications.
+pub mod utils {
+    /// Returns a String containing a pretty printed version of the `buf` slice
+    pub fn hex_dump(buf: &[u8]) -> String {
+        let vec: Vec<String> = buf.iter().map(|b| format!("{:02x}", b)).collect();
+
+        vec.join("")
+    }
+
+    /// ALPN helpers.
+    ///
+    /// This module contains constants and functions for working with ALPN.
+    pub mod alpns {
+        pub const HTTP_09: [&str; 3] = ["hq-24", "hq-23", "http/0.9"];
```

Comment:
Do we need to support hq-23 with this new app?

Reply:
The library supports it, so there doesn't seem harm in allowing it